src/middlewared/middlewared/plugins/smb.py

```diff
@@ -68,6 +68,67 @@ def platform(self):
         return self.value[1] if IS_LINUX else self.value[0]
 
 
+class SMBSharePreset(enum.Enum):
+    DEFAULT = {"verbose_name": "Default share parameters", "params": {
+        'path_suffix': '',
+        'home': False,
+        'ro': False,
+        'browsable': True,
+        'timemachine': False,
+        'recyclebin': False,
+        'abe': False,
+        'hostsallow': [],
+        'hostsdeny': [],
+        'aapl_name_mangling': False,
+        'acl': True,
+        'durablehandle': True,
+        'shadowcopy': True,
+        'streams': True,
+        'fsrvp': False,
+        'auxsmbconf': '',
+    }}
+    ENHANCED_TIMEMACHINE = {"verbose_name": "Multi-user time machine", "params": {
+        'path_suffix': '%U',
+        'auxsmbconf': '\n'.join([
+            'tmprotect:auto_rollback=powerloss',
+            'ixnas:zfs_auto_homedir=true',
+            'ixnas:default_user_quota=1T',
+        ])
+    }}
+    MULTI_PROTOCOL_AFP = {"verbose_name": "Multi-protocol (AFP/SMB) shares", "params": {
+        'acl': True,
+        'aapl_name_mangling': True,
+        'streams': True,
+        'durablehandle': False,
+        'auxsmbconf': '\n'.join([
+            'fruit:locking = netatalk',
+            'fruit:metadata = netatalk',
+            'fruit:resource = file',
+            'streams_xattr:prefix = user.',
+            'streams_xattr:store_stream_type = no',
+            'oplocks = no',
+            'level 2 oplocks = no',
+            'strict locking = auto',
+        ])
+    }}
+    MULTI_PROTOCOL_NFS = {"verbose_name": "Multi-protocol (NFSv3/SMB) shares", "params": {
+        'acl': False,
+        'streams': False,
+        'durablehandle': False,
+        'auxsmbconf': '\n'.join([
+            'oplocks = no',
+            'level 2 oplocks = no',
+            'strict locking = yes',
+        ])
+    }}
+    PRIVATE_DATASETS = {"verbose_name": "Private SMB Datasets", "params": {
```

Comment:
Being nit-picky, but `PRIVATE_DATASETS` is confusing to me because you're turning on feature of `ixnas` related to home directories. Maybe name it something related to home directories?

Reply:
Home directories would make it more confusing. What I need to actually do is deprecate that parameter name (keep it around for backwards-compatibility), and instead call it something like "ixnas:dataset_autocreate". With these changes,  users could potentially have automatic share path creation for things other than username. (For instance, primary group, netbios name of computer, IP address, etc, etc).